src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -370,19 +370,25 @@ public function makePathRelative($endPath, $startPath)
         if ('\\' === DIRECTORY_SEPARATOR) {
             $endPath = str_replace('\\', '/', $endPath);
             $startPath = str_replace('\\', '/', $startPath);
-        }
 
-        $stripDriveLetter = function ($path) {
-            if (strlen($path) > 2 && ':' === $path[1] && '/' === $path[2] && ctype_alpha($path[0])) {
-                return substr($path, 2);
-            }
+            $stripDriveLetter = function ($path) {
+                if (strlen($path) > 2 && ':' === $path[1] && '/' === $path[2] && ctype_alpha($path[0])) {
+                    return array(substr($path, 2), $path[0].$path[1]);
```

Comment:
I wouldn't do this here, you changed what this function does - now name of function is wrong. I suggest to do something like
```php
$isAbsolutePath = function($path) {
    return strlen($path) > 2 && ':' === $path[1] && '/' === $path[2] && ctype_alpha($path[0]);
}

if ($isAbsolutePath($endPath) && $isAbsolutePath($startPath) && substr($endPath, 0, 2) !== substr($startPath, 0, 2)) {
    throw new \InvalidArgumentException("You can't make relative links across drives");
}

$stripDriveLetter = function ($path) (use $isAbsolutePath) {
    if ($isAbsolutePath) {
        return substr($path, 2);
    }

    return $path;
};

$endPath = $stripDriveLetter($endPath);
$startPath = $stripDriveLetter($startPath);
```